server/src/modules/answers/answers.service.ts

```diff
@@ -1,80 +1,32 @@
-import {
-  User as UserModel,
-  Post as PostModel,
-  Agency as AgencyModel,
-  Answer as AnswerModel,
-} from '../../bootstrap/sequelize'
-import { PostStatus, Post } from '~shared/types/base'
-import { Answer } from '../../models'
-import { FindOptions, Model } from 'sequelize'
+import { Answer, PostStatus, Post } from '~shared/types/base'
+import { ModelDef } from '../../types/sequelize'
+import { PostCreation } from '../../models/posts.model'
 
-type AnswerWithRelations = Answer & {
-  postId: number
-  userId: number
-}
+export class AnswersService {
+  private Post: ModelDef<Post, PostCreation>
+  private Answer: ModelDef<Answer>
 
-type PostWithRelations = Model &
-  Post & {
-    getAnswers: (options: FindOptions) => AnswerWithRelations[]
+  constructor({
+    Post,
+    Answer,
+  }: {
+    Post: ModelDef<Post, PostCreation>
+    Answer: ModelDef<Answer>
+  }) {
+    this.Post = Post
+    this.Answer = Answer
   }
 
-type AnswerJSON = Pick<Answer, 'body'> & {
-  user: {
-    displayname: string
-    id: number
-    agency: {
-      logo: string
-    }
-  }
-}
-export class AnswersService {
   /**
    * Returns all answers to a post
    * @param postId id of the post
    * @returns an array of answers
    */
-  listAnswers = async (
-    postId: number,
-  ): Promise<
-    {
-      body: string
-      username: string
-      userId: number
-      agencyLogo: string
-    }[]
-  > => {
-    const post = (await PostModel.findOne({
-      where: { id: postId },
-    })) as PostWithRelations
-    if (post) {
-      const answers = await post.getAnswers({
-        include: [
-          {
-            model: UserModel,
-            include: [AgencyModel],
-          },
-        ],
-      })
-
-      // Redact user info except display name
-      return answers.map((instance) => {
-        const answer = instance.toJSON() as AnswerJSON
-        const {
-          user: {
-            id,
-            displayname,
-            agency: { logo },
-          },
-          ...answerWithoutUser
-        } = answer
-        return {
-          ...answerWithoutUser,
-          username: displayname,
-          userId: id,
-          agencyLogo: logo,
-        }
-      })
-    } else return []
+  listAnswers = async (postId: number): Promise<Answer[]> => {
```

Comment:
Same as above, should this still be returning an array instead of just an Answer object?

Reply:
we'll discuss this on the other comment